src/components/CategorySelector/CategorySelector.js

```diff
@@ -1,48 +1,74 @@
 import React from 'react';
 import {
-  StyleSheet, Text, View, ScrollView,
+  StyleSheet, Text, ScrollView,
 } from 'react-native';
+import { Button } from 'native-base';
+import { arrayOf, func, string } from 'prop-types';
 
-const CatalogScreen = () => (
-  <ScrollView style={styles.root} horizontal>
-    <View style={styles.button}>
-      <Text style={styles.buttonText}>Category</Text>
-    </View>
-    <View style={styles.button}>
-      <Text style={styles.buttonText}>Category</Text>
-    </View>
-    <View style={styles.button}>
-      <Text style={styles.buttonText}>Category</Text>
-    </View>
-    <View style={styles.button}>
-      <Text style={styles.buttonText}>Category</Text>
-    </View>
-    <View style={styles.button}>
-      <Text style={styles.buttonText}>Category</Text>
-    </View>
-    <View style={styles.button}>
-      <Text style={styles.buttonText}>Category</Text>
-    </View>
-  </ScrollView>
-);
+import Colors from '../../constants/Colors';
+import RESOURCE_TYPES from '../../resources/resourceTypes';
+
+const CatalogScreen = ({ categories, selectedCategory, setSelectedCategory }) => {
+  const CategoryButton = ({ category }) => {
+    const categoryDisplay = RESOURCE_TYPES[category];
+    const buttonStyle = category === selectedCategory ? styles.buttonSelected : styles.button;
+    const buttonTextStyle = category === selectedCategory ? styles.buttonSelectedText : null;
+    return (
+      <Button style={buttonStyle} onPress={() => setSelectedCategory(category)}>
+        <Text style={buttonTextStyle}>{categoryDisplay}</Text>
+      </Button>
+    );
+  };
+
+  CategoryButton.propTypes = {
+    category: string.isRequired,
+  };
+
+  return (
+    <ScrollView style={styles.root} horizontal showsHorizontalScrollIndicator={false}>
+      {categories.map((category) => {
+        if (category !== 'Patient') {
```

Comment:
this check should be unnecessary, if `categories` prop was already filtered by `RESOURCE_TYPES`
(e.g.: using `supportedResourcesSelector`, or new selector that is derived from `supportedResourcesSelector`)